src/http/client.cr

```diff
@@ -651,7 +651,11 @@ class HTTP::Client
     socket = @socket
     return socket if socket
 
-    socket = TCPSocket.new @host, @port, @dns_timeout, @connect_timeout
+    # URI host may have ipv6 enclosed by brackets and is fine according to
+    # rfc3986. The host http header must contains the brackets, ie: the same
+    # value as returned by URI#host
+    ip_host = @host[0] == '[' && @host[-1] == ']' ? @host[1..-2] : @host
```

Comment:
`@host.starts_with?('[') && @host.ends_with?(']')` is x2 faster.
`@host.lstrip.rstrip` is said to be 7x than `@host[1..-2]`. I wonder if adding a method to strip both at the start and at the end can make sense.

```crystal
host = "[test]"

Benchmark.ips do |x|
  x.report("index") do
    host[1..-2]
  end

  x.report("strip") do
      host.lstrip.rstrip
  end                              
end

index  29.23M ( 34.21ns) (± 4.66%)  7.72× slower
     strip 225.75M (  4.43ns) (± 3.75%)       fastest
```

Reply:
Thats because `lstrip` doesn't do what you want here?

Reply:
`.lchop('[').rchop(']')` should be shorter here though. Surely the same speed.

Reply:
Sorry for the mistake, I confuse the two. Change `strip` to `chop` to have `lchop.rchop`
With chops the program is slower